pkg/storage/utils/ace/ace.go

```diff
@@ -187,8 +188,34 @@ type ACE struct {
 
 // FromGrant creates an ACE from a CS3 grant
 func FromGrant(g *provider.Grant) *ACE {
+	t := "A"
+	// Currently we only deny the full permission set
+	if grants.PermissionsEqual(&provider.ResourcePermissions{}, g.Permissions) {
+		t = "D"
+		g.Permissions = &provider.ResourcePermissions{
+			AddGrant:             true,
+			Delete:               true,
+			CreateContainer:      true,
+			GetPath:              true,
+			GetQuota:             true,
+			InitiateFileDownload: true,
+			InitiateFileUpload:   true,
+			ListGrants:           true,
+			ListContainer:        true,
+			ListFileVersions:     true,
+			ListRecycle:          true,
+			Move:                 true,
+			RemoveGrant:          true,
+			PurgeRecycle:         true,
+			RestoreFileVersion:   true,
+			RestoreRecycleItem:   true,
+			Stat:                 true,
+			UpdateGrant:          true,
+			DenyGrant:            true,
```

Comment:
Why do we need to set those? Afais they are not used at all?

Reply:
ok, not needed because we only support full denials